### ISSUE TITLE
 Prototype Pollution Vulnerability Affecting requirejs@2.3.6 module

### ISSUE THREAD

**Issue by tariqhawis** [requirejs/r.js#1015]
_Details sent directly to the maintainer_

**Comment by vtulse** [comment]
Team any update on this issues

**Comment by artola** [comment]
@jrburke It would be possible and cheap to sanitize the config, it is only 1 place to fix.

https://github.com/requirejs/r.js/blob/acec5366eb9094e670b6d1a87457634e74d6384e/require.js#L1283

```javascript
function sanitize(obj) {
  return JSON.parse(JSON.stringify(obj));
}

...

            configure: function (cfg) {
               cfg = sanitize(cfg);
```

**Comment by prantlf** [comment]
**EDITED**

Initially, I couldn't reproduce this vulnerability using the [example code](https://gist.github.com/mestrtee/9acae342285bd2998fa09ebcb1e6d30a). When looking at the code, I saw functions `hasProp`, `getOwn`, `eachProp`, `mixin`, which allow only own properties accessed. If you copy own properties from one object to another one, they will not be placed to the prototype.

I couldn't reproduce the vulnerability using [my fork](https://github.com/prantlf/r.js). I didn't notice that I was testing with that version. I could reproduce it using the official require.js 2.3.6.

**Comment by jrburke** [comment]
This should be fixed in 2.3.7: https://github.com/requirejs/requirejs/issues/1854
